generator/testdata/rust/gclient/golden/secretmanager/src/lib.rs

```diff
@@ -301,22 +273,18 @@ impl SecretManagerService {
         &self,
         req: crate::model::ListSecretVersionsRequest,
     ) -> Result<crate::model::ListSecretVersionsResponse> {
-        let query_parameters = [
-            gax::query_parameter::format("pageSize", &req.page_size).map_err(Error::other)?,
-            gax::query_parameter::format("pageToken", &req.page_token).map_err(Error::other)?,
-            gax::query_parameter::format("filter", &req.filter).map_err(Error::other)?,
-        ];
         let client = self.client.inner.clone();
-        let res = client
+        let builder = client
```

Comment:
Should we make this mut and remove the following lets?

Reply:
Something like this does not work:

```rust
        let mut builder = client
            .http_client
            .get(format!("{}/v1/{}/secrets", self.base_path, req.parent,))
            .query(&[("alt", "json")]);
        gax::query_parameter::add(builder, "pageSize", &req.page_size).map_err(Error::other)?;
        gax::query_parameter::add(builder, "pageToken", &req.page_token)
            .map_err(Error::other)?;
        gax::query_parameter::add(builder, "filter", &req.filter).map_err(Error::other)?;
```

The problem is that `builder` gets moved on the first `::add` call.  We could try to pass a `&mut RequestBuilder`, but the `.query()` method expects a value:

https://docs.rs/reqwest/latest/reqwest/struct.RequestBuilder.html#method.query

Open to other ideas if you have them, but I think we can merge this PR and clean this up later?